lib/srv/regular/proxy_test.go

```diff
@@ -86,6 +89,12 @@ func TestParseProxyRequest(t *testing.T) {
 		},
 	}
 
+	server := &Server{
+		hostname:  "redhorse",
+		proxyMode: true,
+		logger:    slog.New(logutils.DiscardHandler{}),
```

Comment:
Should we use a [test logger](https://github.com/gravitational/teleport/blob/5fbb1b522b12d3331f1f4b3301950a8f97cee26d/lib/utils/cli.go#L189-L193) so that it gets the logs when we run the tests in verbose mode (`-v`)?